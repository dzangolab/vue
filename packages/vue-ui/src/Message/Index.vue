<template>
  <div v-show="showMessage" :class="['message', severity]">
    <span
      v-if="icon || !!slots.icon || showIcon"
      :class="['icon', { default: !icon && !slots.icon && showIcon }]"
    >
      <slot v-if="(icon || !!slots.icon) && showIcon" name="icon">
        <i :class="icon" />
      </slot>
    </span>

    <span class="message-content">
      <slot>
        <template v-if="!Array.isArray(message)">
          {{ message }}
        </template>
        <template v-else-if="message.length">
          <ul>
            <li
              v-for="(messageData, index) in message"
              :key="`message-${index}`"
            >
              {{ messageData }}
            </li>
          </ul>
        </template>
      </slot>
    </span>

    <img
      v-if="enableClose"
      src="./../assets/svg/x-mark.svg"
      class="close-message"
      @click="onClose()"
    />
  </div>
</template>

<script lang="ts">
export default {
  name: "Message",
};
</script>

<script setup lang="ts">
import { ref, useSlots } from "vue";

import type { PropType } from "vue";

const emits = defineEmits(["close"]);

defineProps({
  enableClose: {
    default: false,
    type: Boolean,
  },
  icon: {
    default: null,
    type: String,
  },
  message: {
    required: true,
    type: String || (Array as PropType<string[]>),
  },
  severity: {
    type: String,
    default: "info",
    validator: (value: string) =>
      ["danger", "info", "success", "warning"].includes(value),
  },
  showIcon: {
    default: true,
    type: Boolean,
  },
});

const slots = useSlots();

const showMessage = ref<boolean>(true);

const onClose = () => {
  showMessage.value = false;

  emits("close");
};
</script>
