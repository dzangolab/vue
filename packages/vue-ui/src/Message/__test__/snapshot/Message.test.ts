import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";

import Message from "../../Index.vue";

describe("Message", () => {
  it("matches snapshot", () => {
    const wrapper = mount(Message, {
      props: {
        enableClose: true,
        message: "This is a very important message.",
      },
      slots: {
        icon: "<i class='pi pi-bell' />",
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
