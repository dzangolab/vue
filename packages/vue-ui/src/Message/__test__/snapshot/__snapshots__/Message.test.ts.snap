// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Message > matches snapshot 1`] = `
<div
  class="message info"
>
  <span
    class="icon"
  >
    
    <i
      class="pi pi-bell"
    />
    
  </span>
  <div
    class="message-content"
  >
    
    
    This is a very important message.
    
    
  </div>
  <span
    class="close-message"
  />
</div>
`;
